authentication/serializers.py

```diff
@@ -48,3 +56,118 @@ class Meta:
+            "email",
+            "first_name",
+            "last_name",
+            "password",
+            "is_active",
+            "profile",
+        )
+        extra_kwargs = {
+            "password": {
+                "style": {"input_type": "password"},
+                "write_only": True,
+            },
+            "first_name": {
+                "required": True,
+                "write_only": True,
+            },
+            "last_name": {
+                "required": True,
+                "write_only": True,
+            },
+        }
+
+    profile = UserProfileSerializer(write_only=True)
+    is_active = rfs.BooleanField(default=False, read_only=True)
+
+    def validate_profile(self, profile):
+
+        logger.info(f"{profile}")
+
+        self._profile_serializer = UserProfileSerializer(data=profile)
+        self._profile_serializer.is_valid(raise_exception=True)
+        return profile
+
+    def create(self, validated_data):
+        validated_data.pop("profile", None)
+        validated_data["is_active"] = False
+        user = None
+        try:
+            user = super().create(validated_data)
+
+            # save profile object only if user object was actually saved
+            if getattr(user, "pk", None):
+                self._profile_serializer.save(user=user)
+                user.refresh_from_db()
+        except DatabaseError:
+            transaction.rollback()
+        return user
+
+
+class EmailVerificationSerializer(
+    rest_email_auth.serializers.EmailVerificationSerializer
+):
+    def validate_key(self, key):
+        try:
+            return super().validate_key(key)
+        except rfs.ValidationError as exc:
+            # custom error messages
+            err_str = str(exc.detail)
+            if "invalid" in err_str:
+                exc.detail = (
+                    "The provided verification key"
+                    " is invalid or your email address is already verified."
+                )
+            if "expired" in err_str:
+                exc.detail = (
+                    "The provided verification key"
+                    " has expired or your email address is already verified."
+                )
+            raise exc
+
+    def save(self):
+        """
+        Confirm the email address matching the confirmation key.
+        Then mark user as active.
+        """
+        user = self._confirmation.email.user
+        with transaction.atomic():
+            super(EmailVerificationSerializer, self).save()
+
+            # Send msg on slack
+        try:
+            userprofile = user.user_profile
+            user_admin_link = (
+                f"https://gomphidae.prod.srvc.eu.certego.sec/admin/api/user/{user.pk}"
+            )
+            userprofile_admin_link = f"""
+                https://gomphidae.prod.srvc.eu.certego.sec/admin/api/userprofile/{userprofile.pk}
+            """
+            slack = Slack()
+            slack.send_message(
```

Comment:
We should consider the case where the IntelOwl admin does not want to send to slack